WMF Framework/CollectionViewSwipeToEditController.swift

```diff
@@ -0,0 +1,235 @@
+import Foundation
+
+public enum CollectionViewCellSwipeType {
+    case primary, secondary, none
+}
+
+enum CollectionViewCellState {
+    case idle, open
+}
+
+public class CollectionViewSwipeToEditController: NSObject, UIGestureRecognizerDelegate {
+    let collectionView: UICollectionView
+    let panGesture = UIPanGestureRecognizer()
+    let longPressGesture = UILongPressGestureRecognizer()
+    
+    var currentState: CollectionViewCellState = .idle {
+        didSet {
+            currentState == .idle ? didEnterIdleState() : didEnterOpenState()
+        }
+    }
+    
+    var activeCell: ArticleCollectionViewCell? {
+        get {
+            let position = panGesture.location(in: collectionView)
+            let panCellPath = collectionView.indexPathForItem(at: position)
+            if let path = panCellPath, let cell = collectionView.cellForItem(at: path) as? ArticleCollectionViewCell {
+                return cell
+            }
+            return nil
+        }
+        set {
+            
+        }
+    }
+    
+    public var cellWithActionPaneOpen: ArticleCollectionViewCell?
+    
+    public var primaryActions: [CollectionViewCellAction] = []
+    public var secondaryActions: [CollectionViewCellAction] = []
+    
+    fileprivate var theme: Theme = Theme.standard
+    
+    public init(collectionView: UICollectionView, theme: Theme) {
+        self.collectionView = collectionView
+        self.theme = theme
+        super.init()
+        
+        addPanGesture(to: self.collectionView)
+        addLongPressGesture(to: self.collectionView)
+        
+        if let gestureRecognizers = self.collectionView.gestureRecognizers {
```

Comment:
Can this be set up with the `panGestureRecognizer`, `pinchGestureRecognizer`, etc properties? Or, move the addPanGesture, addLongPressGesture to after this (error in the console - it's adding a gesture recognizer as a requirement to fail for itself)